autogluon/task/text_prediction/column_property.py

```diff
@@ -0,0 +1,636 @@
+             ('freq', self.frequencies)])
+
+
+class NumericalColumnProperty(ColumnProperty):
+    type = _C.NUMERICAL
+
+    def __init__(self, shape: Optional[Tuple] = None):
+        """
+
+        Parameters
+        ----------
+        shape
+            The shape of the numerical values
+        """
+        super().__init__()
+        self._shape = shape
+
+    def transform(self, ele):
+        if ele is None:
+            return np.full(shape=self.shape, fill_value=np.nan, dtype=np.float32)
+        else:
+            return np.array(ele, dtype=np.float32)
+
+    @property
+    def shape(self):
+        return self._shape
+
+    def parse(self, column_data: pd.Series):
+        super().parse(column_data)
+        idx = column_data.first_valid_index()
+        val = column_data[idx]
+        inferred_shape = np.array(val).shape
+        if self._shape is not None:
+            assert self._shape == inferred_shape, 'Shape mismatch!. Expected shape={},' \
+                                     ' shape in the dataset is {}'.format(self._shape,
+                                                                          inferred_shape)
+        else:
+            self._shape = inferred_shape
+
+    def clone(self):
+        return NumericalColumnProperty(shape=self.shape)
+
+    def get_attributes(self):
+        return {'shape': self.shape}
+
+    def info(self):
+        return super().info([('shape', self.shape)])
+
+
+class TextColumnProperty(ColumnProperty):
+    type = _C.TEXT
+
+    def __init__(self):
+        super().__init__()
+        self._min_length = None
+        self._max_length = None
+        self._avg_length = None
+
+    @property
+    def min_length(self):
+        return self._min_length
+
+    @property
+    def max_length(self):
+        return self._max_length
+
+    @property
+    def avg_length(self):
+        return self._avg_length
+
+    def parse(self, column_data: pd.Series):
+        super().parse(column_data)
+        lengths = column_data.apply(len)
+        self._min_length = lengths.min()
+        self._avg_length = lengths.mean()
+        self._max_length = lengths.max()
+
+    def clone(self):
+        return TextColumnProperty()
+
+    def get_attributes(self):
+        return {}
+
+    def info(self):
+        return super().info([('length, min/avg/max',
+                              '{:d}/{:.2f}/{:d}'.format(self.min_length,
+                                                        self.avg_length, self.max_length))])
+
+
+def _get_entity_label_type(label) -> str:
+    """
+
+    Parameters
+    ----------
+    label
+        The label of an entity
+
+    Returns
+    -------
+    type_str
+        The type of the label. Will either be null, categorical or numerical
+    """
+    if label is None:
+        return _C.NULL
+    if isinstance(label, (int, str)):
+        return _C.CATEGORICAL
+    else:
+        return _C.NUMERICAL
+
+
+class EntityColumnProperty(ColumnProperty):
```

Comment:
Perhaps it would be good to have an example in the docstring of code to construct this? I am having a hard time following what exactly this is and what it is used for.